### PR TITLE
Feature: add base architecture

### PR DESCRIPTION
**Pull request by iverly**
Hi !

This PR adds the base architecture for the project.

We defined a new cargo project, where each team has its own cargo project. Each team will then restructure their package as they wish, but we needed to define a base structure for the project.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[package]
```
Comment (sameo) [inline Cargo.toml:1]: Here you want to use a workspace instead. That allows for sharing a bunch of things between all crates that are members of the workspace, including external crates dependencies. See https://doc.rust-lang.org/cargo/reference/workspaces.html#the-package-table and https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html for more details. 

Then all other crates are member of that workspace

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "lumper"]
```
Comment (sameo) [inline .gitmodules:1]: I'm fine with submodules, but I want to warn you that working on a submodule and maintaining/submitting changes to lambdo may not be trivial. I expect that you have compared that option with having an actual fork as a local crate. Again, either way is fine with me.

File: src/main.rs [hunk 3]
```diff
@@ -0,0 +1,3 @@
+fn main() {
+    println!("Hello, world!");
+}
```
Comment (sameo) [inline src/main.rs:3]: This does not look like it's up to date.

**Review by sameo — COMMENTED** [review]
Thanks and apologies for the late review. A few comments, and just as a nit: This is not a project architecture but rather a workspace layout.

File: Cargo.toml (hunk 1 above)
Comment (iverly) [reply Cargo.toml:1]: We have chosen to not use workspace. Considering that both the agent and initramfs are completely separate projects (will not directly interact with api & lumper), the choice of not using a workspace seems to us the most judicious.

File: .gitmodules (hunk 2 above)
Comment (iverly) [reply .gitmodules:1]: We had a few words about that and we finally agree that submodules are a good option.

File: src/main.rs (hunk 3 above)
Comment (iverly) [reply src/main.rs:3]: It will be changed on the next pull request. (#11)

File: Cargo.toml (hunk 1 above)
Comment (sameo) [reply Cargo.toml:1]: > We have chosen to not use workspace. Considering that both the agent and initramfs are completely separate projects (will not directly interact with api & lumper), the choice of not using a workspace seems to us the most judicious.

Workspaces do not necessarily contain logically dependant crates. They will share the same output directory, and most importantly the same set of dependent crates. That's quite useful for avoiding dependencies duplication, conflict and making them easier to maintain.

**Review by sameo — APPROVED** [review]

**Pull request merged**